endpoints-control/src/main/java/com/google/api/control/Client.java

```diff
@@ -363,6 +398,51 @@ public void run() {
     }, interval, 1 /* not so high priority */);
   }
 
+  private void flushAndScheduleQuota() {
+    if (resetIfStopped()) {
+      log.log(Level.FINE, "did not schedule quota flush: client is stopped");
+      return;
+    }
+    int interval = quotaAggregator.getFlushIntervalMillis();
+    if (interval < 0) {
+      log.log(Level.FINE, "did not schedule quota flush: caching is disabled");
+      return; // cache is disabled, so no flushing it
+    }
+
+    if (isRunningSchedulerDirectly()) {
+      log.log(Level.FINE, "did not schedule check flush: no scheduler thread is running");
+      return;
+    }
+
+    log.log(Level.FINE, "flushing the quota aggregator");
+    Stopwatch w = Stopwatch.createUnstarted(ticker);
+    List<AllocateQuotaRequest> reqs = quotaAggregator.flush();
+    if (log.isLoggable(Level.FINE)) {
+      log.log(Level.FINE,
+          String.format("flushing %d quota from the quota aggregator", reqs.size()));
+    }
+    for (AllocateQuotaRequest req : reqs) {
+      try {
+        statistics.recachedQuotas.incrementAndGet();
```

Comment:
isn't it the case we haven't actually recached it until the end of this try block? the statistic call might be more accurate there.